From 2144627585b9c61e6659768510b21c285b796398 Mon Sep 17 00:00:00 2001
From: Alvin Wong <alvinhochun@gmail.com>
Date: Wed, 9 Mar 2022 15:12:54 +0800
Subject: [PATCH] Check for uninitialized QDockAreaLayoutInfo

QDockAreaLayoutItem::subinfo may be nullptr if a dock widget had just
been docked with the docking animation still happening.
---
 src/widgets/widgets/qdockarealayout.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/widgets/widgets/qdockarealayout.cpp b/src/widgets/widgets/qdockarealayout.cpp
index 54504d124b..2ab70840aa 100644
--- a/src/widgets/widgets/qdockarealayout.cpp
+++ b/src/widgets/widgets/qdockarealayout.cpp
@@ -1384,6 +1384,10 @@ QRect QDockAreaLayoutInfo::separatorRect(const QList<int> &path) const
     if (path.count() > 1) {
         const QDockAreaLayoutItem &item = item_list.at(index);
         Q_ASSERT(item.subinfo != 0);
+        if (!item.subinfo) {
+            qWarning("QDockAreaLayoutInfo::separatorRect: child item does not have subinfo!");
+            return QRect();
+        }
         return item.subinfo->separatorRect(path.mid(1));
     }
     return separatorRect(index);
-- 
2.24.1.windows.2

